### PR TITLE
Fix problem with entity reference fields

### PR DESCRIPTION
**Pull request by mtodor**

### CODE REVIEW

File: src/ContentCreator.php [hunk 1]
```diff
@@ -383,7 +383,7 @@ protected function createBundle($entity_type, $bundle_type, $num_of_instances, $
         fputcsv($this->cacheCsvFileHandlers[$entity_type][$table_name], array_values($row));
       }
 
-      $this->entityBundleIDs[$entity_type][$bundle_type][] = $total_entity_type_count;
+      $this->entityBundleIDs[$entity_type][$bundle_type][$total_entity_type_count] = $total_entity_type_count;
```
Comment (mtodor) [inline src/ContentCreator.php:386]: We have used the wrong data format here. Function `array_rand` returns keys. 😞

File: src/ContentCreator.php [hunk 2]
```diff
@@ -394,7 +394,7 @@ protected function createBundle($entity_type, $bundle_type, $num_of_instances, $
 
     unset($this->entityTypeReferenceNestingStack[$unique_bundle_key]);
 
-    return range($total_entity_type_count - $num_of_instances + 1, $total_entity_type_count);
+    return range($total_entity_type_count - $num_of_instances, $total_entity_type_count - 1);
```
Comment (mtodor) [inline src/ContentCreator.php:397]: Wrong IDs are generated by this range. We had an offset of 1.

File: src/ContentCreator.php [hunk 3]
```diff
@@ -620,7 +620,7 @@ protected function createEntityReferenceFields($entity_type, $bundle_type, $star
 
         $target_entity_bundles = [];
         $num_of_instances = $this->getRandomWeighted($histogram);
-        for ($instance_index = $num_of_instances; $instance_index >= 0; $instance_index--) {
+        for ($instance_index = 0; $instance_index < $num_of_instances; $instance_index++) {
```
Comment (mtodor) [inline src/ContentCreator.php:623]: This created an additional instance.

**Review by dbosen — APPROVED** [review]
works now, interesting, that the array_rand bug did not provide problems earlier...

**Pull request merged**